Update publication based on JOSS review (#106)
open-pv/simshady: joss-paper/paper.md

  - Energy
  - Solar
  - Photovoltaic
  - Shading Simulation
  - Energy System
authors:

>   - name: Florian Kotthoff

Author: Add TUM affiliation

`simshady` fills a gap in the area of detailed 3D shading simulation for PV yield estimation. This is especially important when modeling PV systems in real environments outside the laboratory, because shading caused by nearby objects, terrain, or other panels significantly lowers the yield of PV systems. Accurate shading simulation is therefore essential for real-world yield estimation.

# State of the field

Existing software can roughly be grouped into three groups: full-suite commercial design tools, simplified yield calculators, and developer libraries.

> Full-suite design tools target professional PV planners and combine 3D system modeling, component databases and financial analysis behind a graphical user interface. HelioScope by Aurora Solar is a web-based design platform that approximates near shading by projecting drop shadows from 3D objects rather than performing full ray tracing, and is offered as a paid subscription [@helioscope2026]. PV\*SOL premium by Valentin Software is a Windows desktop application that performs detailed sub-module shading analysis at cell level, but is restricted to interactive single-project use, meaning it requires manual, GUI-based operation for one project at a time without automation or batch processing capabilities [@pvsol2026]. PVsyst is the industry reference for bankable yield reports and additionally provides a separately licensed command-line tool, PVsystCLI, which is the only existing software that combines accurate 3D near shading with batch automation, but at a cost of several thousand Swiss francs per year [@pvsyst2026]. The System Advisor Model (SAM) developed by NREL is open-source and exposes a Python wrapper (PySAM), but its 3D shading capabilities are limited compared to dedicated design tools [@sam2026].

Reviewer: We have to use the new name now. If you wish, I think it would be acceptable to write "formerly known as..." or something like that.
No need for a hyphen here since "open source" is after the noun. Only needed if it precedes the noun as a compound adjective.

```suggestion
Full-suite design tools target professional PV planners and combine 3D system modeling, component databases and financial analysis behind a graphical user interface. HelioScope by Aurora Solar is a web-based design platform that approximates near shading by projecting drop shadows from 3D objects rather than performing full ray tracing, and is offered as a paid subscription [@helioscope2026]. PV\*SOL premium by Valentin Software is a Windows desktop application that performs detailed sub-module shading analysis at cell level, but is restricted to interactive single-project use, meaning it requires manual, GUI-based operation for one project at a time without automation or batch processing capabilities [@pvsol2026]. PVsyst is the industry reference for bankable yield reports and additionally provides a separately licensed command-line tool, PVsystCLI, which is the only existing software that combines accurate 3D near shading with batch automation, but at a cost of several thousand Swiss francs per year [@pvsyst2026]. The System Advisor Model (SAM) developed by the National Laboratory of the Rockies (NLR) is open source and exposes a Python wrapper (PySAM), but its 3D shading capabilities are limited compared to dedicated design tools [@sam2026].
```

| HelioScope    | Web      | Paid | No          | 3D                    | API       | [@helioscope2026]                        |
| PV\*SOL prem. | Desktop  | Paid | No          | 3D                    | No        | [@pvsol2026]                             |
| PVsyst        | Desktop  | Paid | No          | 3D                    | CLI       | [@pvsyst2026]                            |
| SAM           | Desktop  | Free | Yes         | 3D simplified         | Py        | [@sam2026]                               |
| PVGIS         | Web      | Free | No          | Horizon               | API       | [@pvgis2026]                             |
| PVWatts       | Web      | Free | Yes         | Row                   | API       | [@pvwatts2026]                           |
| SoDeLe        | Package  | Free | Yes         | None                  | Yes       | [@sodele2026]                            |
| pvlib         | Package  | Free | Yes         | Horizon, Row, Diffuse | Yes       | [@holmgren2018pvlib; @anderson2023pvlib] |
| pvfactors     | Package  | Free | Yes         | 2D                    | Yes       | [@pvfactors2025; @anoma_view_2017]       |

> `simshady` now fills two existing gaps: First, it is implemented in TypeScript and hence simultaniously enables simulations in the browser and on a local machine. And second, it tackles the calculation intensive task of raytracing for shading simulation with a performant WebGL implementation on the GPU. The shader language GLSL (OpenGL Shading Language), on which WebGL is built, is not familiar for most developers: it only ranks 35th among programming languages on GitHub [@GitHub2026Innovation]. Hence, `simshady` makes the parallelized simulation of shading more accessible by exposing the WebGL simulation code through the more familiar JavaScript APIs instead.

Reviewer: ```suggestion
`simshady` now fills two existing gaps: First, it is implemented in TypeScript and hence simultaniously enables simulations in the browser and on a local machine. Second, it tackles the calculation intensive task of raytracing for shading simulation with a performant WebGL implementation on the GPU. The shader language GLSL (OpenGL Shading Language), on which WebGL is built, is not familiar for most developers: it only ranks 35th among programming languages on GitHub [@GitHub2026Innovation]. Hence, `simshady` makes the parallelized simulation of shading more accessible by exposing the WebGL simulation code through the more familiar JavaScript APIs instead.
```

# Software design

`simshady` simulates the yield of photovoltaic (PV) systems by considering weather/climate data and shading from local 3D geometry. It is built around three core principles: (1) To run the core simulation code efficiently on the GPU, (2) to expose two different interfaces for both browser-based applications and server-based simulation, and (3) to integrate standardized data models, namely from `Three.js` and from the Wavefront OBJ file format.

## Input data

> In `simshady`, two major types of input data need to be provided. First, a 3D scene is built that represents the environment, consisting of primary objects for simulation (e.g., PV panels or target buildings) and surrounding objects that may cast shadows (e.g., neighboring buildings, trees). These objects can be provided as Wavefront .obj files or as Three.js geometries. Second, weather and climate data need to be provided as input data in .json format by aggregating irradiance data like Global Horizontal Irradiance (GHI) and Direct Normal Irradiance (DNI) datasets onto sky domes, for example by assigning irradiance values to each sky segment [@Górski_2005; @zonca2019healpy].

Reviewer: personal preference, I just think it sounds better. Not super important though
```suggestion
In `simshady`, two major types of input data need to be provided. First, a 3D scene is built that represents the environment, consisting of primary objects for simulation (e.g., PV panels or target buildings) and surrounding objects that may cast shadows (e.g., neighboring buildings, trees). These objects can be provided as Wavefront .obj files or as Three.js geometries. Second, weather and climate data need to be provided as input data in .json format by aggregating irradiance data such as Global Horizontal Irradiance (GHI) and Direct Normal Irradiance (DNI) datasets onto sky domes, for example by assigning irradiance values to each sky segment [@Górski_2005; @zonca2019healpy].
```

The complete simulation took roughly 110 hours on a single cloud GPU server running four parallel CLI instances and cost less than 60 USD. The full output, including geometry, intensities, metadata and visualisation artefacts for all tiles, makes up roughly 35 GB in compressed format and is openly available as a dataset on Zenodo under CC BY 4.0 [@simshady_munich_dataset].

# Conclusion

The `simshady` package serves two primary purposes: it provides a solution for scientific calculations of PV yield on local machines through the CLI, while also facilitating science communication through interactive web-based simulations. The second one eliminates the need for specialized software or programming knowledge, making it accessible to a broader range of users.

Two design decisions are especially relevant: By implementing the main algorithm in WebGL, the package achieves higher performance than a pure JavaScript implementation, and it offers a JavaScript wrapper around PV simulations in WebGL. Additionally, `simshady` simplifies the simulation workflow by processing standardized input formats, thereby simplifying its integration in existing workflows.

> In the future, `simshady` can profit from various developments: First, the architectural decision of outsourcing the calculation of irradiance has a clear benefit. If averaged irradiance data is provided for a whole year, simshady calculates the averaged PV yield per year. If time series of irradiance data are provided, simshady can also simulate time series of PV yield that show the local shading influence at highly resolved times. However, with this benefit comes the disadvantage that users need to provide irradiance data themselves. One useful extension would be to integrate the creation of these time series into the software.

Reviewer: ```suggestion
In the future, `simshady` can profit from various developments: First, the architectural decision of outsourcing the calculation of irradiance has a clear benefit. If averaged irradiance data are provided for a whole year, `simshady` calculates the averaged PV yield per year. If irradiance time series data are provided, `simshady` can also simulate time series of PV yield that show the local shading influence at highly resolved times. However, with this benefit comes the disadvantage that users need to provide irradiance data themselves. One useful extension would be to integrate the creation of these time series into the software.
```
